### ISSUE TITLE
FeatureFactory.AddGraphic incorrect check for range vs single value

### ISSUE THREAD

**Issue by pseudocoder** [HomeSeer/Plugin-SDK#291]
The second IF statement in this block should also check that the statusGraphic is NOT a range, otherwise the single value in a range is checked when it should be ignored.

```
if (statusGraphic.IsRange && _feature.HasGraphicForRange(statusGraphic.TargetRange))
{
    throw new ArgumentException("A value targeted by the specified statusGraphic already has a graphic bound to it.", "statusGraphic");
}

if (_feature.HasGraphicForValue(statusGraphic.Value))
{
    throw new ArgumentException("The value targeted by the specified statusGraphic already has a graphic bound to it.", "statusGraphic");
}
```

suggest replacing with:

```
if (statusGraphic.IsRange)
{
    if (_feature.HasGraphicForRange(statusGraphic.TargetRange))
    {
        throw new ArgumentException("A value targeted by the specified statusGraphic already has a graphic bound to it.", "statusGraphic");
    }
}
else
{
    if (_feature.HasGraphicForValue(statusGraphic.Value))
    {
        throw new ArgumentException("The value targeted by the specified statusGraphic already has a graphic bound to it.", "statusGraphic");
    }
}
```

this also gives clear space for any future range / non-range checking.

**Comment by spudwebb** [comment]
this problem has been fixed in version 1.4.1, check the current code:
https://github.com/HomeSeer/Plugin-SDK/blob/3bacebb4a7bf3fd27e0fd04eb252c323b7d55f06/Devices/FeatureFactory.cs#L703-L708

**Comment by pseudocoder** [comment]
Thanks - I should have checked the version bundled in the sample plugin, which is 1.3.xxx
